### PR TITLE
Breaking change: Add 'list' command to show known databases, rearrange command line options, give better error on unknown db

### PR DESCRIPTION
**Pull request by vinceatbluelabs**
This does three things, in the interest of helping people bootstrap new databases into existence in db-facts:

1. Add 'list' command to show known databases.
2. When a user hands us a database name that doesn't exist, give them a pointer to the configuration guide, and point them to that list command.
3. Rearrange (and break existing command line options) to allow item number 1 to happen.  Today we make the db_name a required positional parameter, which doesn't really work when we have a command which doesn't need a db-name. 

```sh
$ db-facts --help
usage: db-facts [-h] {list,json,config,sh} ...

Pull information about databases from user-friendly names

positional arguments:
  {list,json,config,sh}
    list                List available dbnames
    json                Report output in JSON format
    config              Report output in db-facts config format
    sh                  Report output in Bourne shell envionment variable format

optional arguments:
  -h, --help            show this help message and exit
$ 
```

```
$ db-facts list
Available db_names:
* actionkit
* bltoolsdevbq
* bltoolsdevbq-bq_itest
* bluehq
* cms
* cmshelpdesk
* cmshelpdeskpg
* demo
* demovertica
* devoted
* devotedpg
* devotedsharedpg
* everybodyvotes
* igc
* igcpg
* influencers
* influencerspg-dev
* ivr
* meowx
* meowx-lower
* meowxpicassopg-dev
* meowxpicassopg-sandbox
* mods
* mods-impl
* mods-test
* modspicassopg-impl
* modspicassopg-prod
* modspicassopg-test
* phoenix
* phoenix-blmedia
* phoenix-blmodeling
* phoenix-developer
* picassopg-dev
* picassopg-prod
* redshift
* rippleapppg-dev
* rippleapppg-prod
* rippleapppg-staging
* singularity
* vicalptunnel
$ 
```

```sh
$ db-facts sh some-db-that-doesnt-exit
some-db-that-doesnt-exit is not a valid DB name.  To list valid databases, run "db-facts list" and
to configure a new database, please see https://github.com/bluelabsio/db-facts/blob/master/CONFIGURATION.md
$ 
```

### CODE REVIEW

File: metrics/bigfiles_high_water_mark [hunk 1]
```diff
@@ -1 +1 @@
-326
+328
```
Comment (vinceatbluelabs) [inline metrics/bigfiles_high_water_mark]: runner.py increased in size a touch.  I'm not too worried yet:

```
134: setup.py
111: CONFIGURATION.md
83: db_facts/runner.py

Reduce total number of bigfiles violations to 326 or below!

Tasks: TOP => quality
(See full trace by running task with --trace)
Existing violations: 326
Found 328 bigfiles violations
make: *** [quality] Error 1

Compilation exited abnormally with code 2 at Thu Feb 27 13:24:54
```

**Comment by vinceatbluelabs** [comment]
I'm not too worried about the impact of the breaking change internally; we can hold on releasing this until the mvrec changes also go in.  I don't think a lot of folks use `db-facts` on the command line by hand, and I'll add to the plan putting out compensating PRs to our internal scripts that reference this.

File: metrics/mypy_high_water_mark [hunk 2]
```diff
@@ -1 +1 @@
-47.9200
\ No newline at end of file
+72.6500
```
Comment (cwegrzyn) [inline metrics/mypy_high_water_mark]: 😍

File: setup.cfg [hunk 3]
```diff
@@ -1,3 +1,7 @@
+[coverage:report]
+exclude_lines =
+    if TYPE_CHECKING:
```
Comment (cwegrzyn) [inline setup.cfg]: Huh

File: db_facts/runner.py [hunk 4]
```diff
@@ -15,38 +16,67 @@ def __init__(self) -> None:
 
     def run(self, argv: List[str]) -> int:
         try:
+
+            def dump_json(args: argparse.Namespace) -> None:
```
[the remark this answers is not in the document]
Comment (vinceatbluelabs) [reply db_facts/runner.py]: Nah, not really.  Moved, here as well as in `list_db_names.py`.

**Pull request merged**